yourapp/your_activity_definition.go

```diff
@@ -0,0 +1,50 @@
+// @@@SNIPSTART go-samples-your-activity-definition
+package yourapp
+
+import (
+	"context"
+
+	"go.temporal.io/sdk/activity"
+)
+// Use a struct so that your function signature remains compatible if fields change.
+type YourActivityParam struct {
+	ActivityParamX string
+	ActivityParamY int
+}
+// Use a struct so that you can return multiple values of different types.
+// Additionally, your function signature remains compatible if the fields change.
+type YourActivityResultObject struct {
+	ResultFieldX string
+	ResultFieldY int
+}
+// If the Worker crashes this Activity object loses its state.
+type YourActivityObject struct {
+	SharedMessageState string
+	SharedCounterState int
+}
+// An Activity Definiton is an exportable function.
+func(a *YourActivityObject) YourActivityDefinition(ctx context.Context, param YourActivityParam) (YourActivityResultObject, error) {
+	// Use Acivities for computations or calling external APIs.
+	// This is just an example of appending to text and incrementing a counter.
+	a.SharedMessageState = param.ActivityParamX + " World!"
```

Comment:
This is usually bad as this is not concurrency safe and shared state from an activity should be

Reply:
Shared state field, should it be a pointer then?

Reply:
No, you need to surround it with locks.

Are we expecting to have samples in other languages just like this one? I am curious what the end result may be.